agenthub/micro/agent.py

```diff
@@ -28,32 +26,33 @@ def to_json(obj, **kwargs):
     return json.dumps(obj, **kwargs)
 
 
-def history_to_json(
-    history: ShortTermHistory, max_message_chars: int, max_events: int = 20, **kwargs
-):
-    """
-    Serialize and simplify history to str format
-    """
-    processed_history = []
-    event_count = 0
-
-    for event in history.get_events(reverse=True):
-        if event_count >= max_events:
-            break
-        processed_history.append(event_to_memory(event, max_message_chars))
-        event_count += 1
-
-    # history is in reverse order, let's fix it
-    processed_history.reverse()
-
-    return json.dumps(processed_history, **kwargs)
-
-
 class MicroAgent(Agent):
     VERSION = '1.0'
     prompt = ''
     agent_definition: dict = {}
 
+    def history_to_json(
```

Comment:
Does the prompt.md still work with this being moved inside the class? I haven't tried that.

Reply:
> This is tested in unit tests [here](https://github.com/OpenDevin/OpenDevin/blob/01ce1e35b5b40e57d96b15a7fc9bee4eb8f6966d/tests/unit/test_micro_agents.py#L64), so I assume so.

Oh, I overlooked the `__init__` change below where it is mapped to `self.`.